statsmodels/tsa/tests/test_stattools.py

```diff
@@ -879,13 +879,55 @@ def test_innovations_algo_filter_kalman_filter(reset_randomstate):
     res = mod.filter(np.r_[ar_params, ma_params, sigma2])
 
     # Test that the two approaches are identical
+    assert_allclose(u, res.forecasts_error[0])
+    assert_allclose(theta[1:, 0], res.filter_results.kalman_gain[0, 0, :-1])
+    assert_allclose(llf_obs, res.llf_obs)
     atol = 1e-6 if PLATFORM_WIN else 0.0
     assert_allclose(u, res.forecasts_error[0], atol=atol)
     assert_allclose(theta[1:, 0], res.filter_results.kalman_gain[0, 0, :-1],
                     atol=atol)
     assert_allclose(llf_obs, res.llf_obs, atol=atol)
 
 
+def test_zivot_andrews():
+    cur_dir = os.path.abspath(os.path.dirname(__file__))
+    resdir = os.path.join(cur_dir, "results")
+    zafiles = ['rgnp.csv', 'gnpdef.csv', 'stkprc.csv', 'rgnpq.csv', 'rand10000.csv']
+    for file in zafiles:
+        mdlfile = os.path.join(resdir, file)
+        mdl = np.asarray(pd.read_csv(mdlfile))
+        # compare results to R package urca.ur.za (1.13-0)
+        if file == 'rgnp.csv':
+            res = zivot_andrews(mdl, maxlag=8, regression='c', autolag=None)
+            assert_almost_equal(res[0], -5.57615, decimal=3)
+            assert_almost_equal(res[1], 0.00312, decimal=3)
+            assert_equal(res[4], 20)
+        elif file == 'gnpdef.csv':
+            res = zivot_andrews(mdl, maxlag=8, regression='c', autolag='t-stat')
+            assert_almost_equal(res[0], -4.12155, decimal=3)
+            assert_almost_equal(res[1], 0.28024, decimal=3)
+            assert_equal(res[3], 5)
+            assert_equal(res[4], 40)
+        elif file == 'stkprc.csv':
+            res = zivot_andrews(mdl, maxlag=8, regression='ct', autolag='t-stat')
+            assert_almost_equal(res[0], -5.60689, decimal=3)
+            assert_almost_equal(res[1], 0.00894, decimal=3)
+            assert_equal(res[3], 1)
+            assert_equal(res[4], 65)
+        elif file == 'rgnpq.csv':
+            res = zivot_andrews(mdl, maxlag=12, regression='t', autolag='t-stat')
+            assert_almost_equal(res[0], -3.02761, decimal=3)
+            assert_almost_equal(res[1], 0.63993, decimal=3)
+            assert_equal(res[3], 12)
+            assert_equal(res[4], 102)
+        else:
+            res = zivot_andrews(mdl, regression='c', autolag='t-stat')
+            assert_almost_equal(res[0], -3.48223, decimal=3)
+            assert_almost_equal(res[1], 0.69111, decimal=3)
```

Comment:
im pretty sure assert_allclose is preferred to assert_almost_equal

Reply:
Now using assert_allclose.